<a href="https://colab.research.google.com/github/AshmitaRakshit/Class114/blob/main/Class114_linear_regreshion.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import csv
import pandas as pd
import plotly.express as px

df=pd.read_csv("data.csv")
height=df["Height"].tolist()
weight=df["Weight"].tolist()
fig=px.scatter(x=height, y=weight)
fig.show()

Linear Regression is a machine learning algorithm based on supervised learning. It performs a regression task. Here we can see that both height and weight are related to each other. 
We put the independent variable on the x axis (height).
Linear equation follows a formula where y=mx+c
M is the slope of the line C is the intercept, they are constants.
We will use the hit and trial method to find the method of mc

In [2]:
m = 1
c = 0
y = []
for x in height:
  y_value = m*x + c
  y.append(y_value)

#Plotting the points
fig = px.scatter(x=height, y=weight)
fig.update_layout(shapes=[
    dict(
      type= 'line',
      y0= min(y), y1= max(y),
      x0= min(height), x1= max(height)
    )
])
fig.show()

Let us change the value of m and c as the scatter plot and the line does not fit with each other.

In [3]:
m = 0.95
c = -93
y = []
for x in height:
  y_value = m*x + c
  y.append(y_value)

#Plotting the points
fig = px.scatter(x=height, y=weight)
fig.update_layout(shapes=[
    dict(
      type= 'line',
      y0= min(y), y1= max(y),
      x0= min(height), x1= max(height)
    )
])
fig.show()

Here, we have a close enough line that covers all the scatter plots in our chart.

In [4]:
x = 250
y = m * x + c
print(f"Weight of someone with height {x} is {y}")
#Let us predict the value of weight when the height=250

Weight of someone with height 250 is 144.5


Great! Since we have achieved the values of our slope and intercept with the hit and trial method, we might be a bit less accurate. There are pre-built computer algorithms that give us the value of the slope and the intercept.


Let's find the slope and the intercept of the data that we have using computer algorithms and plot again with the new values of M and C.

[ ]


In [6]:
import numpy as np
height_array = np.array(height)
weight_array = np.array(weight)

#Slope and intercept using pre-built function of Numpy
m, c = np.polyfit(height_array, weight_array, 1)
print(m,c)
y = []
for x in height_array:
  y_value = m*x + c
  y.append(y_value)

#plotting the graph
fig = px.scatter(x=height_array, y=weight_array)
fig.update_layout(shapes=[
    dict(
      type= 'line',
      y0= min(y), y1= max(y),
      x0= min(height_array), x1= max(height_array)
    )
])
fig.show()

0.9600183750530678 -95.66248461433146


The values of m and c from computer algorithm are 0.9600183750530678 -95.66248461433146 

Let us predict the value of weight if height=250

In [8]:
x=250
y=m*x+c
print(y)

144.3421091489355


Weight of someone with height 250 is 144.34 (applying the m and c through algorithm